substratevm/src/com.oracle.svm.core.posix.jdk11/src/com/oracle/svm/core/posix/jdk11/linux/Target_jdk_net_LinuxSocketOptions.java

```diff
@@ -0,0 +1,300 @@
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+package com.oracle.svm.core.posix.jdk11.linux;
+
+import java.net.SocketException;
+
+import com.oracle.svm.core.annotate.Substitute;
+import com.oracle.svm.core.annotate.TargetClass;
+import com.oracle.svm.core.headers.Errno;
+import com.oracle.svm.core.jdk.JDK11OrLater;
+import com.oracle.svm.core.posix.PosixJavaNetClose;
+import com.oracle.svm.core.posix.PosixUtils;
+import com.oracle.svm.core.posix.headers.NetinetIn;
+import com.oracle.svm.core.posix.headers.NetinetTcp;
+import com.oracle.svm.core.posix.headers.Socket;
+import org.graalvm.nativeimage.ImageSingletons;
+import org.graalvm.nativeimage.Platform;
+import org.graalvm.nativeimage.Platforms;
+import org.graalvm.nativeimage.StackValue;
+import org.graalvm.nativeimage.c.struct.SizeOf;
+import org.graalvm.nativeimage.c.type.CIntPointer;
+
+import static com.oracle.svm.core.posix.headers.NetinetTcp.SOL_TCP;
+import static com.oracle.svm.core.posix.headers.NetinetTcp.TCP_KEEPIDLE;
+import static com.oracle.svm.core.posix.headers.NetinetTcp.TCP_KEEPCNT;
+import static com.oracle.svm.core.posix.headers.NetinetTcp.TCP_KEEPINTVL;
+import static com.oracle.svm.core.posix.headers.NetinetTcp.TCP_QUICKACK;
+import static com.oracle.svm.core.posix.headers.Socket.PF_INET;
+import static com.oracle.svm.core.posix.headers.Socket.SOCK_STREAM;
+import static com.oracle.svm.core.posix.headers.Socket.SOL_SOCKET;
+import static com.oracle.svm.core.posix.headers.Socket.getsockopt;
+import static com.oracle.svm.core.posix.headers.Socket.setsockopt;
+import static com.oracle.svm.core.posix.headers.Socket.socket;
+import static com.oracle.svm.core.posix.jdk11.linux.Util_jdk_net_LinuxSocketOptions.handleError;
+import static com.oracle.svm.core.posix.jdk11.linux.Util_jdk_net_LinuxSocketOptions.socketOptionSupported;
+
+@TargetClass(className = "jdk.net.LinuxSocketOptions", onlyWith = JDK11OrLater.class)
+@Platforms({Platform.LINUX.class})
+public final class Target_jdk_net_LinuxSocketOptions {
+
+    // ported from {jdk11}/src/jdk.net/linux/native/libextnet/LinuxSocketOptions.c
+
```

Comment:
Most of the code in this class triggers the formatting checker. Please run the formatting checker locally (which you can do via `mx --strict-compliance eclipseformat -e /path/to/eclipse-4.5.2-linux-amd64/eclipse --primary`) to fix the issues or disable the formatting for these methods.